lib/pbench/server/cache_manager.py

```diff
@@ -444,16 +443,12 @@ def filestream(tarball_path: Path, path: Path) -> Optional[IO[bytes]]:
             return tarfile.open(tarball_path, "r:*").extractfile(path)
         except Exception as exc:
             raise BadDirpath(
-                f"A problem occurred processing {path!r} from {tarball_path!s}: {exc}"
-            ) from exc
-
-    @staticmethod
-    def filecontents(tarball_path: Path, path: Path) -> Optional[str]:
-        """Returns filestream as a string"""
-        return Tarball.filestream(tarball_path, path).read().decode()
+                f"A problem occurred processing {str(path)!r} from {str(tarball_path)!s}: {exc}"
+            )
 
-    def extract(self, path: str) -> Optional[tuple[dict, IO[bytes]]]:
-        """Extracts the contents of a file from the Tarball
+    def filestream(self, path: str) -> Optional[dict]:
+        """Returns the file stream which yields the contents of
+        a file at the specified path in the Tarball
```

Comment:
I think this docstring was meant to be used for `extract()` at line 433.

This function returns a dictionary containing information about the target file, possibly including a stream of its contents.